source/Camera/Basic/C++Halcon/CaptureViaZivid/CaptureViaZivid.cpp

```diff
@@ -19,50 +19,68 @@ HalconCpp::HObjectModel3D zividToHalconPointCloud(const Zivid::PointCloud &point
 	const auto width = pointCloud.width();
 	const auto height = pointCloud.height();
 
-	std::vector<float> pointsX(width * height);
-	std::vector<float> pointsY(width * height);
-	std::vector<float> pointsZ(width * height);
-	std::vector<float> colorsR(width * height);
-	std::vector<float> colorsG(width * height);
-	std::vector<float> colorsB(width * height);
-	std::vector<int> rowMapping(width*height);
-	std::vector<int> colMapping(width*height);
-
-	int numValidPoints = 0;
-	for (size_t i = 0; i < width*height; ++i)
+
+
+	int numberOfValidPoints = std::count_if(pointCloud.dataPtr(), pointCloud.dataPtr() + pointCloud.size(), [](Zivid::Point point)
+	{
+		return (!point.isNaN());
+	});
```

Comment:
Does this introduce a for loop under the hood or is it smarter?

Reply:
It is a lambda, a template for iterating. Lambdas in C++ allows the computer to optimize it if it finds a way to do this, hence it is either as "slow" as a for loop or faster (its never slower). Tested with only counting valid points using lambda and a double for-loop. 

lambda:  median: 9.582 , average: 11.358
for-loop: median: 15.862 , average: 17.816

Reply:
But yeah, it is a for loop